### DISCUSSION TITLE
Song Embedding Solution

### DISCUSSION THREAD

**Discussion by RaafatSalib** [ppy/osu#22590] [Ideas] [answered]
Hello, in the past several months, @wiskerz76 and I have been developing a Spotify client for a mentorship because we thought it would be a really engaging and interesting capstone project for our senior year of high school. We now want to see if we can implement it into osu! such that a beat-map could use any song from Spotify. If possible, we would greatly appreciate it if you all could help us by explaining where and how osu! plays its songs in the codebase by providing an architecture diagram, code walk-through, or even just listing the most applicable files for finding out where music is played from.

**Comment by mk56-spn** [comment]
This isn't going to happen for a plethora of reasons. From being astronomically out of the project's scope, to licensing issues, to the time it would take.. etc..

**Reply by RaafatSalib** [reply]
Just to be clear, we were just hoping to get an architecture diagram. We do not want to commercialize or promote our client; this is just a research project for school. Sorry for the confusion.

**Comment by bdach** [comment, marked as answer]
Doing this would violate [Section III of the Spotify Developer Policy](https://developer.spotify.com/policy/#iii-some-prohibited-applications):

> This section of the Developer Policy identifies some types of SDA and functionality that you must not build, nor enable or encourage others to build:
> (...)
> Do not create a game, including trivia quizzes.

I strongly recommend not to follow through with your plan.
